File: YEAR 2/SEM 1/Advanced Programming Methods/LAB 4/src/View/Command.java
package View;

public abstract class Command {
    private String key,description;

    public Command(String key,String description) {
        this.key = key;
        this.description = description;
    }

    public abstract void execute();
    public String getKey() { return this.key; }
    public String getDescription() { return this.description; }
}

File: YEAR 2/SEM 1/Advanced Programming Methods/ToyLanguage/ToyLanguage/View/Command.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ToyLanguage.View
{
    abstract class Command
    {
        private readonly string key;
        private readonly string description;

        public Command(string key,string description)
        {
            this.key = key;
            this.description = description;
        }

        public abstract void Execute();

        public string Key => key;
        public string Description => description;
    }
}
